modules/OFConnectionManager2/utest/main.c

```diff
@@ -2398,6 +2426,106 @@ test_bad_controller_name(bool use_tls, int domain, char *addr)
            get_domain_name(domain), addr);
 }
 
+static void
+test_cxn_keepalive_max_outstanding_count(bool use_tls)
+{
+    indigo_controller_id_t id;
+    int lsd;
+    intptr_t tl;
+    of_object_t *obj;
+    of_object_storage_t storage;
+    uint8_t buf[512];  /* may need to be increaed */
+
+    printf("***Start %s, %s\n", __FUNCTION__, get_tcp_tls(use_tls));
+
+    /* set up listening socket */
+    lsd = setup_server(AF_INET, CONTROLLER_IP, CONTROLLER_PORT1);
+
+    indigo_setup(use_tls, CIPHER_LIST, CA_CERT_FILE, 
+                 SWITCH_CERT_FILE, SWITCH_PRIV_KEY_FILE, NULL);
+
+    INDIGO_ASSERT((id = setup_cxn_with_echo(use_tls, CONTROLLER_IP, CONTROLLER_PORT1)) >= 0);
+    INDIGO_ASSERT(unit_test_cxn_state_get(id, 0) == CXN_S_INIT);
+
+    OK(ind_soc_select_and_run(1));
+    INDIGO_ASSERT(!cxn_is_connected[id][0]);
+    INDIGO_ASSERT(unit_test_cxn_state_get(id, 0) == CXN_S_HANDSHAKING);
+
+    tl = advance_to_handshake_complete(false, false, id, 0, lsd);
+    INDIGO_ASSERT(unit_test_connection_count_get() == 1);
+
+    printf("cxn socket events %d\n", unit_test_cxn_events_get(id, 0));
+    INDIGO_ASSERT(unit_test_cxn_events_get(id, 0) == POLLIN);
+
+    /* on handshake complete, unsolicited controller_connections_reply is sent */
+    obj = of_recvmsg(use_tls, tl, buf, sizeof(buf), &storage);
+    INDIGO_ASSERT(obj->object_id == OF_BSN_CONTROLLER_CONNECTIONS_REPLY,
+                  "did not receive OF_BSN_CONTROLLER_CONNECTIONS_REPLY");
+    check_connection_list(obj, OF_CONTROLLER_ROLE_EQUAL);
+
+    OK(ind_soc_select_and_run(2000));
+
+    printf("cxn socket events %d\n", unit_test_cxn_events_get(id, 0));
+    INDIGO_ASSERT(unit_test_cxn_events_get(id, 0) == POLLIN);
+
+    /* on handshake complete, unsolicited controller_connections_reply is sent */
+    obj = of_recvmsg(use_tls, tl, buf, sizeof(buf), &storage);
+    INDIGO_ASSERT(obj->object_id == OF_ECHO_REQUEST,
+                  "did not receive OF_ECHO_REQUEST");
+
+    INDIGO_ASSERT(cxn_is_connected[id][0]);
+
+    OK(ind_soc_select_and_run(8000));
```

Comment:
probably add a comment here such as "after 8 seconds, connection should time out" or some similar english